### PR TITLE
Add slack notifications for the jenkins build

### LINKED ISSUE #55: Jenkins job needs to notify on build success or failure
Add a notification that goes to slack and irc to notify us when build succeeds or fails

### PR DESCRIPTION
**Pull request by limed**
Adds slack notifications to developer-portal jenkins builds

(Resolves #55)

## Key changes:

- Adding slack notifications for the jenkins build pipeline

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -36,14 +61,26 @@ node {
   switch (env.BRANCH_NAME) {
     case 'master':
       stage ('Buld Image') {
```
Comment (stevejalim) [inline Jenkinsfile]:
```suggestion
      stage ('Build Image') {
```

Comment (limed) [reply Jenkinsfile]: Fixed :)

**Comment by limed** [comment]
> I was thinking that you might want to add, within the `deploy` function, a call to `notify_slack` on success for a positive indication of success rather than just an absence of a failure messsage, but that depends on what you all want. Otherwise everything looks great!

I think thats a good point, I went ahead and added that

File: Jenkinsfile [hunk 2]
```diff
@@ -16,13 +27,32 @@ def deploy(config) {
...
+      sh """
+        . config/${config}.sh
+        export APP_IMAGE_TAG=${tag}
+        make k8s-db-migration-job
+        make k8s-deployments
+        make k8s-rollout-status
+      """
+      notify_slack([
+        stage: "Deployed to ${environment}",
+        status: 'sucess'
```
Comment (escattone) [inline Jenkinsfile]:
```suggestion
        status: 'success'
```

Comment (limed) [reply Jenkinsfile]: Fixed in f839bab874e44aea0e3f4d82e3123cf588b5ec04

**Pull request merged**